pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleTypedProducerConsumerTest.java

```diff
@@ -459,7 +461,7 @@ public void testAvroProducerAndAutoSchemaConsumer() throws Exception {
        }
 
        Consumer<GenericRecord> consumer = pulsarClient
-           .newConsumer(Schema.AUTO())
+           .newConsumer(Schema.AUTO_CONSUME())
```

Comment:
why are these methods, and not fields? It looks weird.

Reply:
I made them methods to be consistent with AVRO and JSON.

Reply:
AVRO and JSON take arguments though. A static method that takes no arguments is effectively a constant, and if it's not, it's doing something very suspicious in the background.

Anyhow, I wouldn't hold up the change for it, but I think it would be better as a constant.

Reply:
actually it can't be a constant. because it needs to set SchemaInfo when the client fetches schema info from brokers.

Reply:
same applied to both `AUTO_CONSUME()` and `AUTO_PRODUCE`